### PR TITLE
The eventgenerator should not send trigger alarm when the appmetrics …

### PR DESCRIPTION
**Pull request by qibobo**
…of an application  do not cover a whole breach duration.

[[#156261863]]

### CODE REVIEW

File: src/autoscaler/eventgenerator/generator/evaluator_test.go [hunk 1]
```diff
@@ -315,26 +155,59 @@ var _ = Describe("Evaluator", func() {
...
 					Context("when the appMetrics breach the trigger", func() {
-						BeforeEach(func() {
-							database.RetrieveAppMetricsStub = func(appId string, metricType string, start int64, end int64) ([]*models.AppMetric, error) {
-								return appMetricGTBreach, nil
-							}
-						})
-						It("should send trigger alarm to scaling engine", func() {
-							Eventually(scalingEngine.ReceivedRequests).Should(HaveLen(1))
-							Eventually(logger.LogMessages).Should(ContainElement(ContainSubstring("send trigger alarm to scaling engine")))
+						Context("when the appMetrics are enough", func() {
```
Comment (boyang9527) [inline src/autoscaler/eventgenerator/generator/evaluator_test.go]: we do not need to test not enough metrics for each operator.  The test case with not enough metrics should not be under "when the appMetrics brach the trigger"

**Comment by qibobo** [comment]
@boyang9527 updated.

**Comment by boyang9527** [comment]
I am thinking perhaps we can just retrieve metrics in 2*aggregation_interval + breach_duration  as opposed to 2* breach_duration, this will prevent too many metrics if breach_duration is very large.. Anyway this could be another PR for optimization

**Comment by cdlliuy** [comment]
LGTM

**Pull request merged**